src/App.tsx

```diff
@@ -15,69 +15,141 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+type State = {
+  isReversed: boolean;
+  sortType: SortType;
+};
+
 enum SortType {
   NONE,
   ALPHABET,
   LENGTH,
 }
 
-type ReorderOptions = {
-  sortType: SortType;
-  isReversed: boolean;
-};
-
-// Use this function in the render method to prepare goods
 export function getReorderedGoods(
   goods: string[],
-  { sortType, isReversed }: ReorderOptions,
+  { sortType, isReversed }: State,
 ) {
-  // To avoid the original array mutation
   const visibleGoods = [...goods];
 
-  // Sort and reverse goods if needed
-  // eslint-disable-next-line no-console
-  console.log(sortType, isReversed);
+  if (sortType === SortType.ALPHABET) {
+    visibleGoods.sort((a: string, b: string) => a.localeCompare(b));
+  } else if (sortType === SortType.LENGTH) {
+    visibleGoods.sort((a: string, b: string) => a.length - b.length);
+  }
+
+  if (isReversed) {
+    visibleGoods.reverse();
+  }
 
   return visibleGoods;
 }
 
-// DON'T save goods to the state
-// type State = {
-//   isReversed: boolean,
-//   sortType: SortType,
-// };
-
-export const App: React.FC = () => {
-  return (
-    <div className="section content">
-      <div className="buttons">
-        <button type="button" className="button is-info is-light">
-          Sort alphabetically
-        </button>
-
-        <button type="button" className="button is-success is-light">
-          Sort by length
-        </button>
-
-        <button type="button" className="button is-warning is-light">
-          Reverse
-        </button>
-
-        <button type="button" className="button is-danger is-light">
-          Reset
-        </button>
-      </div>
+export class App extends React.Component {
+  state: State = {
+    isReversed: false,
+    sortType: SortType.NONE,
+  };
+
+  resetGoodsSorting = () => {
+    this.setState((state: State) => {
+      return {
+        ...state,
+        isReversed: false,
+        sortType: SortType.NONE,
+      };
+    });
+  };
+
+  sortGoodsAlphabet = () => {
+    this.setState((state: State) => {
+      return {
+        ...state,
+        sortType: SortType.ALPHABET,
+      };
+    });
+  };
+
+  sortGoodsByLength = () => {
+    this.setState((state: State) => {
+      return {
+        ...state,
+        sortType: SortType.LENGTH,
+      };
+    });
+  };
+
+  handleReverseGoods = () => {
+    this.setState((state: State) => {
+      return {
+        ...state,
+
+        isReversed: !state.isReversed,
+      };
+    });
+  };
+
+  render() {
+    const { sortType, isReversed } = this.state;
+
+    const reorderedGoods = getReorderedGoods(goodsFromServer, {
+      sortType: sortType,
+      isReversed: isReversed,
+    });
+
+    function goodsSortedOrReversed(
```

Comment:
Better to call it 'displayResetButton' - it will just provide more clarity for functionality behind this function. 
Btw, just to go a bit fancy, can't it be an arrow function?